### PR TITLE
[JN-1075] external link support

### PR DESCRIPTION
**Pull request by devonbush**
#### DESCRIPTION (include screenshots, and mobile screenshots for participant UX)
This adds the ability to add external link navbar items via the site content editor
image
When an external link is selected, the href is editable
image

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

1. go to https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/siteContent
2. try adding/deleting navabar items of both internal and external types

### CODE REVIEW

**Review by MatthewBemis — APPROVED** [review]
Looks good, just a small testing note: if you put in `google.com` without any https prefix, the navbar link will send you to `https://sandbox.ourhealth.localhost:3001/google.com` instead

File: ui-admin/src/portal/siteContent/SiteContentEditor.tsx [hunk 1]
```diff
@@ -286,45 +299,64 @@ const SiteContentEditor = (props: InitializedSiteContentViewProps) => {
...
+                    <HtmlSectionView section={section} key={section.id}/>)
+                  }
+                  <SiteFooter footerSection={localContent.footerSection}/>
+                </ApiProvider>
+              </ErrorBoundary>
+            </Tab>
+          </Tabs> }
+        { currentNavBarItem?.itemType === 'EXTERNAL' && <div className="mt-2">
+          <form>
+            <label htmlFor="externalHref">External link</label>
```
Comment (connorlbark) [inline ui-admin/src/portal/siteContent/SiteContentEditor.tsx:344]: nit: some padding (`px-2` or `ps-2`) & a fixed size so it isn't huge on a big monitor would look nice.

**Comment by devonbush** [comment]
good note, Matt.  I'll update the help text to clarify that

File: ui-core/src/types/landingPageConfig.ts [hunk 2]
```diff
@@ -52,9 +52,11 @@ export type NavbarItem =
...
 
-type BaseNavBarItem = {
+export type NavBarItemType = 'INTERNAL' | 'INTERNAL_ANCHOR' | 'MAILING_LIST' | 'EXTERNAL'
+
+export type BaseNavBarItem = {
   id?: string
-  itemType: 'INTERNAL' | 'INTERNAL_ANCHOR' | 'MAILING_LIST' | 'EXTERNAL'
+  itemType: NavBarItemType
   text: string
   itemOrder: number
```
Comment (connorlbark) [inline ui-core/src/types/landingPageConfig.ts:61]: do we also want control over whether or not it opens in a new page?

**Review by connorlbark — APPROVED** [review]
I couldn't get it to work in demo because saving it deleted all the multi-language stuff, but that's out of the scope of this. It worked in ourhealth, though, so just some optional thoughts!

File: ui-core/src/types/landingPageConfig.ts (hunk 2 above)
Comment (devonbush) [reply ui-core/src/types/landingPageConfig.ts:61]: that's a good idea, I'll defer that for a later ticket

**Pull request merged**